pkg/chunk/cache/memcached_client.go

```diff
@@ -92,13 +92,17 @@ func NewMemcachedClient(cfg MemcachedClientConfig, name string) MemcachedClient
 	client.Timeout = cfg.Timeout
 	client.MaxIdleConns = cfg.MaxIdleConns
 
+	registerer := prometheus.WrapRegistererWithPrefix("cortex_", prometheus.WrapRegistererWith(prometheus.Labels{
```

Comment:
I would just suggest to rename `registerer` to `dnsProviderRegisterer` to avoid in the future people will erroneously use the replaced `registerer` to register other metrics.